keps/sig-api-machinery/0006-apply.md

```diff
@@ -145,11 +148,58 @@ What are some important details that didn't come across above.
 Go in to as much detail as necessary here.
 This might be a good place to talk about core concepts and how they releate.
 
+#### API Topology
+
+Server-side apply has to understand the topology of the objects in order to make
+valid merging decisions. In order to reach that goal, some new Go markers, as
+well as OpenAPI extensions have been created:
+
+##### Lists
+
+Lists can behave in mostly 3 different ways depending on what their actual semantic
+is. New annotations allow API authors to define this behavior.
+
+- Atomic lists: The list is owned by only one person and can only be entirely
+replaced. This is the default for lists. It is defined either in Go IDL by
+pefixing the list with `// +listType=atomic`, or in the OpenAPI
+with `"x-kubenetes-list-type": "atomic"`.
+
+- Sets: the list is a set (it has to be of a scalar type). Items in the list
+must appear at most once. Individual actors of the API can own individual items.
+It is defined either in Go IDL by pefixing the list with `//
++listType=set`, or in the OpenAPI with
+`"x-kubenetes-list-type": "set"`.
+
+- Associative lists: Kubernetes has a pattern of using lists as dictionary, with
+"name" being a very common key. People can now reproduce this pattern by using
+`// +listType=associative`, or in the OpenAPI with
+`"x-kubenetes-list-type": "map"` along with `//
++x-kubernetes-list-map-key=name`, or `"listMapKey": "name"`. Items
+of an associative lists are owned by the person who applied the item to the
+list.
```

Comment:
We've left out "lists where the index is a reasonable key" -- I think we'll eventually want this, but not right now. (not asking for a change)